src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-snapshotschedule-list/cephfs-snapshotschedule-list.component.ts

```diff
@@ -0,0 +1,100 @@
+import { Component, Input, OnChanges, OnInit, SimpleChanges, ViewChild } from '@angular/core';
+import { NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { finalize, shareReplay, switchMap } from 'rxjs/operators';
+import { CephfsSnapshotScheduleService } from '~/app/shared/api/cephfs-snapshot-schedule.service';
+import { CdForm } from '~/app/shared/forms/cd-form';
+import { CdTableAction } from '~/app/shared/models/cd-table-action';
+import { CdTableColumn } from '~/app/shared/models/cd-table-column';
+import { CdTableFetchDataContext } from '~/app/shared/models/cd-table-fetch-data-context';
+import { CdTableSelection } from '~/app/shared/models/cd-table-selection';
+import { Permissions } from '~/app/shared/models/permissions';
+import { SnapshotSchedule } from '~/app/shared/models/snapshot-schedule';
+import { RelativeDatePipe } from '~/app/shared/pipes/relative-date.pipe';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { ModalService } from '~/app/shared/services/modal.service';
+import { Icons } from '~/app/shared/enum/icons.enum';
+
+@Component({
+  selector: 'cd-cephfs-snapshotschedule-list',
+  templateUrl: './cephfs-snapshotschedule-list.component.html',
+  styleUrls: ['./cephfs-snapshotschedule-list.component.scss']
+})
+export class CephfsSnapshotscheduleListComponent extends CdForm implements OnInit, OnChanges {
+  @Input() fsName!: string;
+
+  @ViewChild('pathTpl', { static: true })
+  pathTpl: any;
+
+  snapshotSchedules$!: Observable<SnapshotSchedule[]>;
+  subject$ = new BehaviorSubject<SnapshotSchedule[]>([]);
+  isLoading$ = new BehaviorSubject<boolean>(true);
+  columns: CdTableColumn[] = [];
+  tableActions: CdTableAction[] = [];
+  context!: CdTableFetchDataContext;
+  selection = new CdTableSelection();
+  permissions!: Permissions;
+  modalRef!: NgbModalRef;
+  errorMessage: string = '';
+  selectedName: string = '';
+  icons = Icons;
+
+  constructor(
+    private snapshotScheduleService: CephfsSnapshotScheduleService,
+    private authStorageService: AuthStorageService,
+    private modalService: ModalService,
+    private relativeDatePipe: RelativeDatePipe
+  ) {
+    super();
+    this.permissions = this.authStorageService.getPermissions();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.fsName) {
+      this.subject$.next([]);
+    }
+  }
+
+  ngOnInit(): void {
+    this.snapshotSchedules$ = this.subject$.pipe(
+      switchMap(() =>
+        this.snapshotScheduleService
+          .getSnapshotScheduleList('/', this.fsName)
+          .pipe(finalize(() => this.isLoading$.next(false)))
+      ),
+      shareReplay(1)
+    );
+
+    this.columns = [
+      { prop: 'path', name: $localize`Path`, flexGrow: 3, cellTemplate: this.pathTpl },
```

Comment:
@nizamial09 
If there's a way to achieve this without a custom template I would prefer but I am doing a bit more here because I am also showing the status of the snapshot with icons.

Reply:
@ivoalmeida ah okay, got you now. thanks. Then I would be okay with it. We should show the full path as a tooltip btw. Similar [to this](https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.html#L348)

Reply:
done